target_parquet/helpers.py

```diff
@@ -70,22 +99,62 @@ def flatten_schema(dictionary, parent_key="", sep="__"):
                             }
                         }
                     }
-    By calling the function with the dictionary above as parameter, you will get the following strucure:
-        [
-             'key_1',
-             'key_2__key_3',
-             'key_2__key_4__key_5',
-             'key_2__key_4__key_6'
-        ]
+    By calling the function with the dictionary above as parameter, you will get the following structure:
+        {
+             'key_1': ['null', 'integer'],
+             'key_2__key_3': ['null', 'string'],
+             'key_2__key_4__key_5': ['null', 'integer'],
+             'key_2__key_4__key_6': ['null', 'array']
+        }
     """
-    items = []
+    items = {}
     if dictionary:
-        for k, v in dictionary.items():
-            new_key = parent_key + sep + k if parent_key else k
-            if "type" not in v:
-                LOGGER.warning(f"SCHEMA with limitted support on field {k}: {v}")
-            if "object" in v.get("type", []):
-                items.extend(flatten_schema(v.get("properties"), new_key, sep=sep))
+        for key, value in dictionary.items():
+            new_key = parent_key + sep + key if parent_key else key
+            if "type" not in value:
+                LOGGER.warning(f"SCHEMA with limited support on field {key}: {value}")
+            if "object" in value.get("type", []):
+                items.update(flatten_schema(value.get("properties"), new_key, sep=sep))
             else:
-                items.append(new_key)
+                items[new_key] = value.get("type", None)
     return items
+
+
+def _field_type_to_pyarrow_field(field_name: str, input_types: Union[List[str], str]):
```

Comment:
Move FIELD_TYPE_TO_PYARROW close to this function maybe? It will be clearer where this constant is needed for.